src/driver/safe/core.rs

```diff
@@ -194,7 +194,11 @@ impl<T> Drop for CudaSlice<T> {
     fn drop(&mut self) {
         self.device.bind_to_thread().unwrap();
         unsafe {
-            result::free_async(self.cu_device_ptr, self.device.stream).unwrap();
+            if self.device.is_async().unwrap() {
+                result::free_async(self.cu_device_ptr, self.device.stream).unwrap();
+            } else {
+                // result::free_sync(self.cu_device_ptr).unwrap();
```

Comment:
Intended to be commented out?

Reply:
Sorry, I was trying things out because I still see some illegal address on compute_cap 5.2.
No it needs to be there !